Core/Src/main.c

```diff
@@ -87,17 +97,26 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
-  /* USER CODE BEGIN 2 */
+  MX_SPI1_Init();
+  MX_TIM1_Init();
 
+  /* USER CODE BEGIN 2 */
+  HAL_TIM_PWM_Init(&htim1); //start PWM
```

Comment:
I think this needs to be HAL_TIM_PWM_Start and with the htim and the channel as arguments.